### DISCUSSION TITLE
Help with nested queries

### DISCUSSION THREAD

**Discussion by BinaryVigilante** [sidorares/node-mysql2#2311] [Q&A] [answered]
Hello,

I'm currently building a discord.js bot and have up until now been using sqlite for my database but has just now moved to the mysql2 library, however I'm running into an issue:

```js
databaseConnection.query("SELECT * FROM table WHERE row1 = ?", row1, (err, result) => {
        if (err) {
            return console.error(err)
        }

        if (!result[0]) {
            databaseConnection.query("INSERT INTO table (row1, row2) VALUES (?,?)", row1,row2, (err) => {
                if (err) {
                    return console.error(err)
                }
               console.log("Success")
            })
        }
    })
```
However it has come to my attention that you aren't able to actually run this code because this library (mysql2) *requires* async and promises?

I have a few questions:
1. Why is that necessary when it can be done without the use of promises and async using sqlite *without issues*
2. Is there a workaround so I don't have to use async and promises?
3. Is the above code supposed to work or am I doing something wrong?

**Comment by sidorares** [comment, marked as answer]
> Why is that necessary when it can be done without the use of promises and async using sqlite without issues

likely your driver exposes synchronous api. mysql2 is built on top of node `net` functionality and only provides async interface.

> Is there a workaround so I don't have to use async and promises?

you could try
https://www.npmjs.com/package/sync-mysql

or wrap your code as an executable script and call it with `execSync`
I'd imagine performance is going to be much worse with sync versions

> Is the above code supposed to work or am I doing something wrong?

pass your parameters as an array: `databaseConnection.query("INSERT INTO table (row1, row2) VALUES (?,?)", row1,row2, (err) => {`. Single parameter might work as is but I prefer to use arrays even for one parameter: `databaseConnection.query("SELECT * FROM table WHERE row1 = ?", [row1], (err, result) => {`

**Comment by BinaryVigilante** [comment]
@sidorares Thank you so much! You lead me in the right direction, I was able to use `execSync` and it works perfectly. I have been searching for days, thank you so much!
